### PR TITLE
add --no-os-network-stats-reporting option

### PR DESCRIPTION
**Pull request by jbiseda**
#### Problem
Add ability to bypass OS network stats reporting. Useful for running under WSL.

#### Summary of Changes
add --no-os-network-stats-reporting option

Fixes #
#21253

### CODE REVIEW

**Review by jeffwashington — APPROVED** [review]
lgtm

File: core/src/validator.rs [hunk 1]
```diff
@@ -449,11 +451,16 @@ impl Validator {
...
-            abort();
-        });
-        let system_monitor_service = Some(SystemMonitorService::new(Arc::clone(&exit)));
+        if !config.no_os_network_stats_reporting {
+            verify_udp_stats_access().unwrap_or_else(|err| {
+                error!("Failed to access UDP stats: {}. Bypass check with --no-os-network-stats-reporting.", err);
+                abort();
+            });
+        }
+        let system_monitor_service = Some(SystemMonitorService::new(
```
Comment (mvines) [inline core/src/validator.rs:460]: Why start the service at all when `config.no_os_network_stats_reporting` is false?

Comment (jeffwashington) [reply core/src/validator.rs:460]: We could also turn off memory reporting, copy the code, or move memory reporting elsewhere.

**Review by mvines — APPROVED** [review]

File: core/src/validator.rs (hunk 1 above)
Comment (mvines) [reply core/src/validator.rs:460]: I think it's ok as is, was just asking

**Pull request merged**